src/webserver/webcache/WebLruCache.java

```diff
@@ -0,0 +1,195 @@
+      }
+  
+      this.lookup.put(id, newNode);
+    }
+  }
+
+  /**
+   * Updates a cached item.
+   * <p>
+   * The object stored under the existing id will be swapped with the new item. If
+   * the id does not exist, nothing happens.
+   * <p>
+   * This method is similar to calling {@link #putCache(String, String, int)} with
+   * a replacement object, but using this method indicates an update. This cached
+   * object will also retain the existing amount of seconds to live from the old
+   * object.
+   * 
+   * @param id      the id to update.
+   * @param newItem the new item to replace the old item.
+   */
+  public synchronized void updateCache(String id, String newItem) {
+    synchronized (this.lookup) {
+      if (checkCache(id)) {
+        this.lookup.get(id).data.updateObject(newItem);
+      }
+    }
+  }
+
+  /**
+   * Attempts to retrieve a cached object using a specified key.
+   * <p>
+   * If the cached object and key does not exist, {@code null} will be
+   * returned.
+   * 
+   * @param id the id of the cached object to fetch.
+   * @return the cached object, or {@code null}.
+   */
+  public synchronized String getCachedObject(String id) {
+    synchronized (this.lookup) {
+        return this.lookup.get(id).data.getObject();
+    }
+  }
+
+  /**
+   * Checks the cache for a specified id.
+   * 
+   * @param id the id to check the cache for.
+   * @return true if the cache has the specified id.
+   */
+  public synchronized boolean checkCache(String id) {
+    synchronized (this.lookup) {
+      return this.lookup.containsKey(id);
+    }
+  }
+
+  /**
+   * Clears the cache and removes all cached objects.
+   */
+  public void clearCache() {
+    this.lookup.clear();
+  }
+
+  /**
+   * Initializes the clean-up service, which will run on a low priority background
+   * thread, routinely cleaning up the cache.
+   */
+  private void initializeCleanupThread() {
+    Thread cleanupThread = new Thread(new Runnable() {
+      public synchronized void run() {
+        while (runCleanup) {
+          synchronized (lookup) {
+            for (String toCheck : lookup.keySet()) {
+              if (lookup.get(toCheck).data.isExpired()) {
+                // Remove expired cache objects
+                Node<TimedObject<String>> toRemove = lookup.get(toCheck);
+                lookup.remove(toCheck);
+  
+                if (toRemove.next != null) {
+                  toRemove.next.prev = toRemove.prev;
+                }
+  
+                if (toRemove.prev != null) {
+                  toRemove.prev.next = toRemove.next;
+                }
+  
+                if (toRemove == head) {
+                  head = toRemove.next;
+                }
+  
+                if (toRemove == tail) {
+                  tail = toRemove.prev;
+                }
+              }
+            }
+          }
+        }
+      }
+    });
+
+    cleanupThread.setPriority(Thread.MIN_PRIORITY);
+    cleanupThread.start();
+  }
```

Comment:
Do you think that marking this thread as MIN_PRIORITY magically makes it pause for a few seconds between each loop? You're effectively locking the map over and over in an infinite loop, increasing the time the other methods need to wait for a chance to lock the map because now there is this loop trying to lock the map every chance it gets. Even after you change to a ConcurrentHashMap and don't need to lock to read the map, please add a 10 second delay or something longer.